src/python/cli_new/lib/cli/config.py

```diff
@@ -119,6 +119,65 @@ def master(self):
 
         return master
 
+    def principal(self):
+        """
+        Return the principal in the configuration file
+        """
+        if "principal" not in self.data["master"]:
+            return None
```

Comment:
I add the check if the parameter set and not empty. Do u have other checks in your mind? :-)

Reply:
Expanding on @asekretenko suggestion, this might be better:
```
principal = self.data["master"].get("principal")
if principal is None:
    return None
elif not principal:
    raise CLIException("The 'principal' field in 'master'"
                                   " must be non-empty")
return principal
```

Same thing for the other accessors.

Reply:
Hm, you don't seem to have done this?

Reply:
I didn't feel it's a so big different to before. Thats why I ask! But I have no problem to change it. What do u think?

Reply:
Ok, I change it. :-) Shell I squash the commit again? :joy:

Reply:
Actually, I can also remove the Exception handling. The principal, and secret is not a "must-have".